src/main/java/thestonedturtle/mahoganyhomes/MahoganyHomesPlugin.java

```diff
@@ -327,6 +349,16 @@ public void onChatMessage(ChatMessage e)
 		}
 	}
 
+	@Subscribe
+	public void onItemContainerChanged(ItemContainerChanged event)
+	{
+		if (event.getContainerId() == InventoryID.INVENTORY.getId())
```

Comment:
Only do this is they currently have an active contract and the plugin hasn't "timed out". Many people leave the plugin on at all times even if they aren't actively doing the content.